clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1023,7 +1024,7 @@ public void assign(Collection<TopicPartition> partitions) {
             return records;
 
         fetcher.sendFetches();
-        client.poll(timeout, now);
+        client.poll(Math.min(maxPollTimeoutMs, timeout), time.milliseconds());
```

Comment:
Is `maxPollTimeoutMs` actually the right choice here? We need to invoke frequently enough, but this is *right* at the limit if the other timeout is large. Should we divide by 2 or something?